runtime/src/bank/fee_distribution.rs

```diff
@@ -70,26 +83,32 @@ impl Bank {
             fee_budget_limits.prioritization_fee,
             FeeFeatures::from(self.feature_set.as_ref()),
         );
-        let (reward, _burn) =
-            self.calculate_reward_and_burn_fee_details(&CollectorFeeDetails::from(fee_details));
+        let BurnedTransactionFee {
+            deposit: reward,
+            burn: _,
+        } = self.calculate_reward_and_burn_fee_details(&CollectorFeeDetails::from(fee_details));
         reward
     }
 
     fn calculate_reward_and_burn_fee_details(
         &self,
         fee_details: &CollectorFeeDetails,
-    ) -> (u64, u64) {
-        let (deposit, burn) = if fee_details.transaction_fee != 0 {
-            self.burn_transaction_fee(fee_details.transaction_fee)
-        } else {
-            (0, 0)
-        };
-        (deposit.saturating_add(fee_details.priority_fee), burn)
+    ) -> BurnedTransactionFee {
+        if fee_details.transaction_fee == 0 {
```

Comment:
Out of scope:

What is the purpose of this `if`? for tests? Will tests break if we remove this?

Reply:
just a path for early exist.